src/Interpreters/TelemetryCollector.cpp

```diff
@@ -1,78 +1,95 @@
 #include "TelemetryCollector.h"
 #include "config_version.h"
 
-#include <Poco/Util/AbstractConfiguration.h>
+#include <filesystem>
+#include <Core/ServerUUID.h>
+#include <IO/WriteBufferFromString.h>
+#include <IO/WriteHelpers.h>
+#include <Interpreters/Context.h>
+#include <base/ClockUtils.h>
+#include <base/getMemoryAmount.h>
 #include <Poco/Net/HTTPRequest.h>
 #include <Poco/Net/HTTPResponse.h>
 #include <Poco/Net/HTTPSClientSession.h>
-#include <base/ClockUtils.h>
-#include <base/getMemoryAmount.h>
+#include <Poco/Util/AbstractConfiguration.h>
 #include <Common/DateLUT.h>
 #include <Common/getNumberOfPhysicalCPUCores.h>
-#include <Core/ServerUUID.h>
-#include <IO/WriteBufferFromString.h>
-#include <IO/WriteHelpers.h>
-#include <Interpreters/Context.h>
-#include <filesystem>
 
 namespace fs = std::filesystem;
 
 namespace ProfileEvents
 {
-    extern const Event SelectQuery;
-    extern const Event StreamingSelectQuery;
-    extern const Event HistoricalSelectQuery;
+extern const Event SelectQuery;
+extern const Event StreamingSelectQuery;
+extern const Event HistoricalSelectQuery;
 }
 
 namespace DB
 {
+namespace
+{
+constexpr auto DEFAULT_INTERVAL_MS = 5 * 60 * 1000;
+}
 
 TelemetryCollector::TelemetryCollector(ContextPtr context_)
-    : log(&Poco::Logger::get("TelemetryCollector")),
-    pool(context_->getSchedulePool()),
-    started_on_in_minutes(UTCMinutes::now())
+    : log(&Poco::Logger::get("TelemetryCollector")), pool(context_->getSchedulePool()), started_on_in_minutes(UTCMinutes::now())
 {
     const auto & config = context_->getConfigRef();
 
-    if (!config.getBool("telemetry_enabled", true))
```

Comment:
I didn't read the whole code. Can we change this setting while server is running and take effect? Can be a SQL or a REST API. The user case is to dynamically turn on/off telemetry without restarting the server.